python/ray/resource_spec.py

```diff
@@ -229,12 +229,31 @@ def resolve(self, is_head, node_ip_address=None):
 def _autodetect_num_gpus():
     """Attempt to detect the number of GPUs on this machine.
 
-    TODO(rkn): This currently assumes Nvidia GPUs and Linux.
+    TODO(rkn): This currently assumes NVIDIA GPUs on Linux.
+    TODO(mehrdadn): This currently does not work on macOS.
+    TODO(mehrdadn): Use a better mechanism for Windows.
+
+    Possibly useful: tensorflow.config.list_physical_devices()
 
     Returns:
         The number of GPUs if any were detected, otherwise 0.
     """
-    proc_gpus_path = "/proc/driver/nvidia/gpus"
-    if os.path.isdir(proc_gpus_path):
-        return len(os.listdir(proc_gpus_path))
-    return 0
+    result = 0
+    if sys.platform.startswith("linux"):
+        proc_gpus_path = "/proc/driver/nvidia/gpus"
+        if os.path.isdir(proc_gpus_path):
+            result = len(os.listdir(proc_gpus_path))
+    elif sys.platform == "win32":
+        props = "AdapterCompatibility"
+        cmdargs = ["WMIC", "PATH", "Win32_VideoController", "GET", props]
+        lines = subprocess.check_output(cmdargs).splitlines()[1:]
+        result = len([l.rstrip() for l in lines if l.startswith(b"NVIDIA")])
+    else:
+        tf = None
+        try:
+            import tensorflow as tf  # This is slow & requires CUDA libraries.
+        except ImportError:
+            pass
+        if tf:
+            result = len(tf.config.list_physical_devices("GPU"))
```

Comment:
In what situation does this work? Also, I think this may break many existing applications because ray is sensitive to GPU imports; I would remove this.

Reply:
Oh I see. I think it _should_ work for Windows, but I can't check right now because I don't have the appropriate setup for it handy. I can take it out if you'd like. But currently this is only for Mac, where we always `return 0` currently, so I figured it's better than nothing. If you think it might cause issues I can take it out.

Reply:
Yeah; let's take it out - thanks!

Reply:
Okay thanks, just did.